src/v2/providers/alerts/alerts.ts

```diff
@@ -0,0 +1,127 @@
+import { ManifestEndpoint } from '../../../common/manifest';
+import { CloudEvent, CloudFunction } from '../../core';
+import * as options from '../../options';
+
+/** Data */
+export interface FirebaseAlertData<T = any> {
+  createTime: string;
+  endTime: string;
+  payload: T;
+}
+
+/** @internal */
+export const eventType = 'firebase.firebasealerts.alerts.v1.published';
+
+/** The underlying alert type of the Firebase Alerts provider */
+export type AlertType =
+  | 'crashlytics.newFatalIssue'
+  | 'crashlytics.newNonfatalIssue'
+  | 'crashlytics.regression'
+  | 'crashlytics.stabilityDigest'
+  | 'crashlytics.velocity'
+  | 'crashlytics.newAnrIssue'
+  | 'billing.planUpdate'
+  | 'billing.automatedPlanUpdate'
+  | 'appDistribution.newTesterIosDevice'
+  | string;
+
+/** Options */
+export interface FirebaseAlertOptions extends options.EventHandlerOptions {
+  alertType: AlertType;
+  appId?: string;
+}
+
+/** Cloud Event Type */
+interface WithAlertTypeAndApp {
+  alertType: string;
+  appId?: string;
+}
+export type AlertEvent<T> = CloudEvent<
+  FirebaseAlertData<T>,
+  WithAlertTypeAndApp
+>;
+
+/** Handlers */
+/** Handle an alert published */
+export function onAlertPublished<T extends { ['@type']: string } = any>(
+  alertTypeOrOpts: AlertType | FirebaseAlertOptions,
+  handler: (event: AlertEvent<T>) => any | Promise<any>
+): CloudFunction<FirebaseAlertData<T>> {
+  const [opts, alertType, appId] = getOptsAndAlertTypeAndApp(alertTypeOrOpts);
+
+  const func = (raw: CloudEvent<unknown>) => {
+    return handler(
+      raw as CloudEvent<FirebaseAlertData<T>, WithAlertTypeAndApp>
+    );
+  };
+
+  func.run = handler;
+
+  // TypeScript doesn't recognize defineProperty as adding a property and complains
+  // that __endpoint doesn't exist. We can either cast to any and lose all type safety
+  // or we can just assign a meaningless value before calling defineProperty.
+  func.__trigger = 'silence the transpiler';
+  func.__endpoint = {} as ManifestEndpoint;
```

Comment:
Would prefer

```
func.__endpoint = defineEndpoint(...)
```

since I think we should shy away from type assertion if it's unnecessary.

Just inlining the function implementation here could also work fine?

Reply:
Re-named and changed to

```
func.__endpoint = getEndpointAnnotation(...);
```